core/iwasm/common/wasm_runtime_common.c

```diff
@@ -3616,6 +3871,11 @@ wasm_externref_obj2ref(WASMModuleInstanceCommon *module_inst, void *extern_obj,
     ExternRefMapNode *node;
     uint32 externref_idx;
 
+    if (NULL_REF == (uintptr_t)extern_obj) {
```

Comment:
Seems we should redefine the NULL_REF for 32-bit and 64-bit:
```C
#if UINTPTR_MAX == UINT32_MAX
#define NULL_REF (0xFFFFFFFF)
#else
#define NULL_REF (0xFFFFFFFFFFFFFFFFLL)
#endif
```

Reply:
`NULL_REF` is used inside. It should always be the same size with a reference index length. 32bit for NON-GC building.